src/js/components/DataTable/buildState.js

```diff
@@ -141,22 +141,26 @@ const groupData = (nextProps, prevState, nextState) => {
 
   let groups;
   let groupState;
+  let expandedState;
   if (groupBy) {
     groups = [];
     groupState = {};
     const groupMap = {};
     data.forEach(datum => {
-      const groupValue = datum[groupBy];
+      const groupByProperty = groupBy.property ? groupBy.property : groupBy;
+      const groupValue = datum[groupByProperty];
       if (!groupMap[groupValue]) {
         const group = { data: [], datum: {}, key: groupValue };
-        group.datum[groupBy] = groupValue;
+        group.datum[groupByProperty] = groupValue;
         groups.push(group);
-        groupState[groupValue] = {
-          expanded:
-            prevState.groupState && prevState.groupState[groupValue]
-              ? prevState.groupState[groupValue].expanded
-              : false,
-        };
+        if (prevState.groupState && prevState.groupState[groupValue]) {
```

Comment:
I think we should switch this around such that if the caller provides an `expand`, then that takes precedence. If it isn't provided, then the previous state is used. With this approach, a caller can control the expansion state entirely, using `expand` and `onChangeExpand`.  If the caller doesn't want to control it, she can omit `expand`. So, if you want to just set the initial expansion state but then not control it afterwards, you can provide `expand` initially but then omit it afterwards.

Reply:
Thanks, this part was updated as well as story and doc.js